crates/noirc_evaluator/src/ssa/acir_gen/internal_var_cache.rs

```diff
@@ -76,10 +98,185 @@ impl InternalVarCache {
             .expect("ICE: `NodeId` type cannot be converted into an `InternalVar`")
     }
 
-    pub(super) fn update(&mut self, id: NodeId, var: InternalVar) {
-        self.inner.insert(id, var);
-    }
     pub(super) fn get(&mut self, id: &NodeId) -> Option<&InternalVar> {
         self.inner.get(id)
     }
+
+    // Transform a field element into a witness
+    // It implements a 'get or create' pattern to ensure only one witness is created per element
+    fn const_to_witness(
+        &mut self,
+        value: FieldElement,
+        evaluator: &mut Evaluator,
+        ctx: &SsaContext,
+    ) -> Witness {
+        if let Some(ids) = ctx.constants.get(&value) {
+            //we have a constant node object for the value
+            if !ids.is_empty() {
+                let id = ids.first().unwrap();
```

Comment:
Instead of checking if `ids` is empty and then unwrap, we can call `ids.first()` and then either do `if let Some(id) = ids.first() {}` or a match

Reply:
Done